packages/TorneloScoresheet/src/pages/EditMove/GraphicalEditMove.tsx

```diff
@@ -0,0 +1,258 @@
+import React, { useEffect, useRef, useState } from 'react';
+import { View } from 'react-native';
+import { ScrollView } from 'react-native-gesture-handler';
+import ActionBar from '../../components/ActionBar/ActionBar';
+import { ActionButtonProps } from '../../components/ActionButton/ActionButton';
+import ChessBoard from '../../components/ChessBoard/ChessBoard';
+import MoveCard from '../../components/MoveCard/MoveCard';
+import OptionSheet from '../../components/OptionSheet/OptionSheet';
+import { useEditMoveState } from '../../context/AppModeStateContext';
+import {
+  BISHOP,
+  ICON_FLIP,
+  ICON_SKIP,
+  ICON_UNDO,
+  KNIGHT,
+  PAWN,
+  QUEEN,
+  ROOK,
+} from '../../style/images';
+import { PlayerColour } from '../../types/ChessGameInfo';
+import {
+  PieceType,
+  MoveSquares,
+  ChessPly,
+  Move,
+  PlyTypes,
+} from '../../types/ChessMove';
+import { colours } from '../../style/colour';
+import { useError } from '../../context/ErrorContext';
+import GraphicalModePlayerCard from '../../components/GraphicalModePlayerCard/GraphicalModePlayerCard';
+import { isError } from '../../types/Result';
+import { styles } from './style';
+import IconButton from '../../components/IconButton/IconButton';
+
+const GraphicalEditMove: React.FC = () => {
+  const editMoveStateHook = useEditMoveState();
+  const editMoveState = editMoveStateHook?.[0];
+
+  const [flipBoard, setFlipBoard] = useState(
+    editMoveState?.currentPlayer === PlayerColour.Black,
+  );
+
+  const [, showError] = useError();
+  const [showPromotion, setShowPromotion] = useState(false);
+  const editingMove = editMoveState?.moveHistory[editMoveState.editingIndex];
+  const editingMoveSquares =
+    editingMove && editingMove.type === PlyTypes.MovePly
+      ? editingMove.move
+      : undefined;
+
+  // Scroll view ref
+  const scrollRef = useRef<ScrollView>(null);
+
+  // when the promotion popup opens, the app will await untill a promise is resolved
+  // this ref stores this resolve function (it will be called once the user selects a promotion)
+  const promotionSelectedFunc = useRef<
+    ((value: PieceType | PromiseLike<PieceType>) => void) | null
+  >(null);
+
+  // Button parameters
+  const actionButtons: ActionButtonProps[] = [
+    {
+      text: 'restart',
```

Comment:
This actually wraps on an iPad because the word is too wide for the button. Maybe "reset" would be better? It conveys the same meaning but in less characters